client/src/components/Recipe.js

```diff
@@ -14,42 +15,49 @@ class Recipe extends Component {
   }
 
   onUpvote(e) {
-    this.props.upvoteRecipe(this.props.recipe.id,this.props.index);
+    console.log(this, this.props);
+    this.props.upvoteRecipe(this.props.recipe.id, this.props.index);
```

Comment:
You can deconstruct the props to avoid long lines. `const { recipe, index, upvoteRecipe } = this.props;`

You can now do `upvoteRecipe(recipe.id, index)`